include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -630,20 +645,80 @@ Vinecop::get_threshold() const
   return threshold_;
 }
 
+//! @brief sets variable types.
+//! @param var_types a vector of size two specifying the types of the variables,
+//!   e.g., `{"c", "d"}` means first varible continuous, second discrete.
+inline void
+Vinecop::set_var_types(const std::vector<std::string>& var_types)
+{
+  std::stringstream msg;
+  if (var_types.size() > d_) {
+    msg << "more var_types (" << var_types.size() << ")"
+        << "than variables (" << d_ << ")-" << std::endl;
+  }
```

Comment:
I would throw the error here already, and then throw another one if the type is incorrect, otherwise the message could look very weird no?

Reply:
Agree